main/src/ca/uwaterloo/flix/language/phase/Parser2.scala

```diff
@@ -1586,92 +1593,104 @@ object Parser2 {
-             | TokenKind.Plus
-             | TokenKind.TripleTilde
-             | TokenKind.KeywordLazy
-             | TokenKind.KeywordForce
-             | TokenKind.KeywordDiscard => unaryExpr()
-        case TokenKind.KeywordIf => ifThenElseExpr()
-        case TokenKind.KeywordLet => letMatchExpr()
-        case TokenKind.Annotation | TokenKind.KeywordDef => localDefExpr()
-        case TokenKind.KeywordRegion => scopeExpr()
-        case TokenKind.KeywordMatch => matchOrMatchLambdaExpr()
-        case TokenKind.KeywordTypeMatch => typematchExpr()
-        case TokenKind.KeywordChoose
-             | TokenKind.KeywordChooseStar => restrictableChooseExpr()
-        case TokenKind.KeywordForA => forApplicativeExpr()
-        case TokenKind.KeywordForeach => foreachExpr()
-        case TokenKind.KeywordForM => forMonadicExpr()
-        case TokenKind.CurlyL => blockOrRecordExpr()
-        case TokenKind.ArrayHash => arrayLiteralExpr()
-        case TokenKind.VectorHash => vectorLiteralExpr()
-        case TokenKind.ListHash => listLiteralExpr()
-        case TokenKind.SetHash => setLiteralExpr()
-        case TokenKind.MapHash => mapLiteralExpr()
-        case TokenKind.DotDotDot => dotdotdotLiteral()
-        case TokenKind.KeywordCheckedCast => checkedTypeCastExpr()
-        case TokenKind.KeywordCheckedECast => checkedEffectCastExpr()
-        case TokenKind.KeywordUncheckedCast => uncheckedCastExpr()
-        case TokenKind.KeywordUnsafe => unsafeExpr()
-        case TokenKind.KeywordUnsafely => unsafelyRunExpr()
-        case TokenKind.KeywordRun => runExpr()
-        case TokenKind.KeywordHandler => handlerExpr()
-        case TokenKind.KeywordTry => tryExpr()
-        case TokenKind.KeywordThrow => throwExpr()
-        case TokenKind.KeywordNew => ambiguousNewExpr()
-        case TokenKind.KeywordStaticUppercase => staticExpr()
-        case TokenKind.KeywordSelect => selectExpr()
-        case TokenKind.KeywordSpawn => spawnExpr()
-        case TokenKind.KeywordPar => parYieldExpr()
-        case TokenKind.HashCurlyL => fixpointConstraintSetExpr()
-        case TokenKind.HashParenL => fixpointLambdaExpr()
-        case TokenKind.KeywordSolve => fixpointSolveExpr()
-        case TokenKind.KeywordInject => fixpointInjectExpr()
-        case TokenKind.KeywordQuery => fixpointQueryExpr()
-        case TokenKind.BuiltIn => intrinsicExpr()
-        case TokenKind.LiteralStringInterpolationL
-             | TokenKind.LiteralDebugStringL => interpolatedStringExpr()
-        case TokenKind.KeywordDebug
-             | TokenKind.KeywordDebugBang
-             | TokenKind.KeywordDebugBangBang => debugExpr()
-        case t =>
-          val mark = open()
-          val error = UnexpectedToken(expected = NamedTokenSet.Expression, actual = Some(t), SyntacticContext.Expr.OtherExpr, loc = currentSourceLocation())
-          advance()
-          closeWithError(mark, error)
+      val token = nth(0)
+      if (exitTokens.contains(token)) {
```

Comment:
Exactly my thinking yes, but I would only do it for the case `case _ =>` error case. The difference is mostly philosophical.

The meaning would be `if you find something that you do not expect - exit and give an error if it is in "exitTokens" - continue looking if it is not in "exitTokens"`. The last part is not relevant here but i hope it makes some sense